specs/098-self-service-email-change/data-model.md

```
@@ -0,0 +1,268 @@
+CREATE INDEX ix_email_change_pending_terminal_updated
+  ON email_change_pending (state, updated_date)
+  WHERE state IN ('rolled_back', 'expired', 'superseded', 'drift_detected', 'committed');
+```
+
+**Why the partial unique on `(subject_user_id) WHERE state IN ('initiated', 'confirmed')`**: This is the database-level mechanism enforcing FR-019 (idempotent re-initiation) and FR-007(d) (supersession). A second `initiated` row for the same user fails the unique constraint; the service-layer flow transitions the existing row to `superseded` and INSERTs the new one. Terminal-state rows do not occupy the slot, so a user with a `committed` pending change from yesterday can still start a fresh one today.
+
+**Why the unique on `token`**: Defence in depth. Token entropy (FR-007c) makes a collision astronomically unlikely, but the unique index ensures we cannot accidentally issue duplicate tokens.
+
+### State transitions (self-initiated flow)
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)